DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc

```diff
@@ -0,0 +1,467 @@
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "LocalPositionXY_S");
+    psd0.add<std::string>("title", "LocalPositionXY_S;x ;y ;");
+    psd0.add<int>("NxBins", 50);
+    psd0.add<double>("xmin", -10.0);
+    psd0.add<double>("xmax", 10.0);
+    psd0.add<int>("NyBins", 50);
+    psd0.add<double>("ymin", -10.0);
+    psd0.add<double>("ymax", 10.0);
+    psd0.add<bool>("switch", true);
+    desc.add<edm::ParameterSetDescription>("LocalPositionXY_S", psd0);
+  }
+
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "CurvatureOfVecHits");
+    psd0.add<std::string>("title", ";VectorHit curvature;");
+    psd0.add<double>("xmin", -0.05);
+    psd0.add<bool>("switch", true);
+    psd0.add<double>("xmax", 0.05);
+    psd0.add<int>("NxBins", 200);
+    desc.add<edm::ParameterSetDescription>("Curvature", psd0);
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "CurvatureErrOCurvature");
+    psd0.add<std::string>("title", ";VectorHit #delta#rho/#rho;");
+    psd0.add<double>("xmin", -0.05);
+    psd0.add<bool>("switch", true);
+    psd0.add<double>("xmax", 0.05);
+    psd0.add<int>("NxBins", 500);
+    desc.add<edm::ParameterSetDescription>("CurvErr", psd0);
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "PhiOfVecHits");
+    psd0.add<std::string>("title", ";VectorHit #phi;");
+    psd0.add<double>("xmin", -M_PI);
+    psd0.add<bool>("switch", true);
+    psd0.add<double>("xmax", M_PI);
+    psd0.add<int>("NxBins", 30);
+    desc.add<edm::ParameterSetDescription>("Phi", psd0);
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "EtaOfVecHits");
+    psd0.add<std::string>("title", ";VectorHit #eta;");
+    psd0.add<double>("xmin", -5.);
+    psd0.add<bool>("switch", true);
+    psd0.add<double>("xmax", 5.);
+    psd0.add<int>("NxBins", 50);
+    desc.add<edm::ParameterSetDescription>("Eta", psd0);
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "CurvatureVsEtaProf_P");
+    psd0.add<std::string>("title", "Curvature vs #eta (macro-pixel);#eta ;curvature ;");
+    psd0.add<int>("NxBins", 50);
+    psd0.add<double>("xmin", -5.0);
+    psd0.add<double>("xmax", 5.0);
+    psd0.add<double>("ymin", -0.05);
+    psd0.add<double>("ymax", 0.05);
+    psd0.add<bool>("switch", true);
+    desc.add<edm::ParameterSetDescription>("CurvatureVsEta_P", psd0);
+  }
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("name", "CurvatureVsEtaProf_S");
+    psd0.add<std::string>("title", "Curvature vs #eta (strip);#eta ;curvature ;");
+    psd0.add<int>("NxBins", 25);
+    psd0.add<double>("xmin", -5.0);
+    psd0.add<double>("xmax", 5.0);
+    psd0.add<double>("ymin", -0.05);
+    psd0.add<double>("ymax", 0.05);
+    psd0.add<bool>("switch", true);
+    desc.add<edm::ParameterSetDescription>("CurvatureVsEta_S", psd0);
+  }
```

Comment:
I am missing plots for some of the quantities one can observe, namely:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L81

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L82

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L82

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L88

Reply:
Thanks! Yes chi2, momentum can be added. The dimension is always 4, that's why it has not been added.

Reply:
> The dimension is always 4, that's why it has not been added.

mmmh, indeed

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L139

wondering what purpose it serves...